src/query/storage/promremote/storage.go

```diff
@@ -95,11 +98,12 @@ func (wq *WriteQueue) Add(query *storage.WriteQuery) []*storage.WriteQuery {
 	// but the majority of the time it won't be full and therefore not worth optimizating.
 	// NB: we have to check if the queue is full under the lock. Otherwise, two goroutines
 	// may see the full queue and try to pop it at the same time.
+	var res []*storage.WriteQuery
 	if len(wq.queries) >= wq.capacity {
-		return wq.popUnderLock()
+		res = wq.popUnderLock()
 	}
 	wq.queries = append(wq.queries, query)
-	return nil
+	return res
```

Comment:
fyi @hczhu-db @yuchen-db @christopherzli , this is the actual bug, this will drop the last wq when the queue is full, 1 out of 1000 chance :)